text/0000-api-version-compat.md

```diff
@@ -0,0 +1,119 @@
+
+# Motivation
+[motivation]: #motivation
+
+## Non-Breaking 0.x API Versions
+Originally we specified that all minor version increments in the `0.x` line would be interpreted as breaking changes, when applying [compatibility rules](https://github.com/buildpacks/spec/blob/master/buildpack.md#buildpacktoml-toml).
+This was the correct decision initially b/c we needed a way to indicate breaking changes before the APIs reached 1.0.
+However, this has become annoying over time as it adds considerable friction to the release of non-breaking features. All platforms and buildpacks must explicitly update to support the new APIs before pulling in a new lifecycle, even when a new minor would not break them.
+As the community of platform and buildpack authors grows and matures, this pain is magnified.
+
+Now that the APIs have stabilized we can realistically guarantee that we will not make breaking lifecycle changes before the implementation of the 1.0 APIs are released.
+
+## Deprecations
+We will allow the spec to remove features in a minor API and interperet this as a deprecated feature in the lifecycle.
+Previous we never formalized rules around deprecations. This made sense because all API changes have been breaking.
+However, deprecations can aid platform and buldpack as they prepare to perform a major API upgrade.
+
+The lifecycle will support deprecation modes to allow platform/buildpack authors to curate the experience for users
+and/or filter for deprecations relevant to their use case.
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Spec
+The spec will use future `0.x` API release to indicate changes that can be implemented in a non-breaking fashion. These may include:
+- purely additive features
+- simple flag/file/key renames (the previous usage will be supported but deprecated by the lifecycle)
+- removal of features (these will be interpreted as deprecated by the lifecycle)
+
+## Lifecycle
+The lifecycle will continue use the [lifecycle descriptor](https://github.com/buildpacks/rfcs/blob/pack-publish-buildpack/text/0011-lifecycle-descriptor.md) 
+to indicate the implemented API versions as before. But the assumptions we make about compatibility will change.
+
+## Deprecations
+If a feature is removed from the lifecycle's advertised API version but used by a platform or buildpack the lifecycle will print a deprecation warning.
+Because platform authors may want to shield end users from deprecation warning caused by the platform's use of deprecated
+lifecycle features, the lifecycle will provide the following options to configure deprecation warnings:
+
+All lifecycle binaries will have a deprecation mode, configurable in the following way:
+| Flag              | Environment Variable | Default
+|-------------------|----------------------|--------
+| -deprecation-mode | CNB_DEPRECATION_MODE | all
+
+The allowed values of deprecation mode are as follows:
+| Mode      | Behavior
```

Comment:
should there be a mode where it's errors (and exits non-zero) instead of warns when a deprecated feature is used? I wonder if this is important for automated tests that are future-proofing.

Reply:
added